### PR TITLE
Don't add additional CR LF to the end of a message when sending

### PR DESCRIPTION
**Pull request by cketti**
This is an alternative version of PR #1371

### CODE REVIEW

File: k9mail-library/src/main/java/com/fsck/k9/mail/filter/EOLConvertingOutputStream.java [hunk 1]
```diff
@@ -7,8 +7,10 @@
 public class EOLConvertingOutputStream extends FilterOutputStream {
     private static final int CR = '\r';
     private static final int LF = '\n';
-    private int lastChar;
-    private static final int IGNORE_LF = Integer.MIN_VALUE;
+
+
+    private int[] lastBytes = new int[2];
```
Comment (Valodim) [inline k9mail-library/src/main/java/com/fsck/k9/mail/filter/EOLConvertingOutputStream.java]: I had to stop and ponder what the order of these bytes might be. Maybe just use `lastByte` and `secondLastByte` to be more clear, or add a comment about its order.

Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/filter/EOLConvertingOutputStream.java]: I think we can get away with only remembering the last byte. I'll update the PR accordingly.

**Comment by cketti** [comment]
Updated.

**Comment by Valodim** [comment]
lgtm

**Pull request merged**